controllers/component_build_controller.go

```diff
@@ -226,28 +209,43 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 		log.Info("Starting Pipelines as Code provision for the Component")
 
+		var pacAnnotationValue string
 		done, err := r.ProvisionPaCForComponent(ctx, &component)
-		if err != nil {
+		if !done {
+			// Need to retry
+			if err != nil {
+				log.Error(err, "Pipelines as Code provision transient error")
+			}
 			return ctrl.Result{}, err
 		}
-		var pacAnnotationValue string
-		if done {
-			pacAnnotationValue = PaCProvisionDoneAnnotationValue
-			log.Info("Pipelines as Code provision for the Component finished successfully")
-		} else {
+		if err != nil {
 			pacAnnotationValue = "error"
 			log.Error(err, "Pipelines as Code provision for the Component failed")
+		} else {
+			pacAnnotationValue = PaCProvisionDoneAnnotationValue
+			log.Info("Pipelines as Code provision for the Component finished successfully")
 		}
 
-		// Update PaC annotation to show that PaC provision is done
+		// Update component to show Pipeline as Code provision is done
 		if err := r.Client.Get(ctx, req.NamespacedName, &component); err != nil {
 			log.Error(err, "failed to get Component")
 			return ctrl.Result{}, err
 		}
+
+		// Update PaC annotation
 		if len(component.Annotations) == 0 {
 			component.Annotations = make(map[string]string)
 		}
 		component.Annotations[PaCProvisionAnnotationName] = pacAnnotationValue
+
+		// Add finalizer to clean up Pipelines as Code configuration on component deletion
+		if component.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
Another thought, is it worth to make a separate function to examine whether a resource is being deleted? Something like:
```go
func isResourceBeingDeleted() bool {
...
}
if isResourceBeingDeleted(component) {

}
```

Reply:
We can do that, but the construction above is a commonly used pattern